### DISCUSSION TITLE
トラックボールが動作しない

### DISCUSSION THREAD

**Discussion by torikatsupg** [Yowkees/keyball#276] [Q&A]
keyball39のビルドガイドを参考に組み立てたのですが、トラックボールのみ動作しません。
過去のQ&A(特に以下2つ)を参考にはんだ付けのやり直し等してみましたが、
OLEDに表示された移動量はすべて0のままでした。
- https://github.com/Yowkees/keyball/discussions/274#discussion-5400367
- https://github.com/Yowkees/keyball/discussions/51#discussioncomment-1044412

他に原因として、どのようなことがことが考えられるでしょうか。

| マイコン裏側  | L字ピンヘッダ  | センサー表面  | センサー裏面  |
|:-:|:-:|:-:|:-:|
| ![IMG_4194](https://github.com/Yowkees/keyball/assets/50566499/42044f34-b812-4449-a3a1-bbc9fa295709) | ![IMG_4195](https://github.com/Yowkees/keyball/assets/50566499/293d4c1a-fce3-4042-abb0-6086597e7ef1) | ![IMG_4196](https://github.com/Yowkees/keyball/assets/50566499/87977e51-65fb-406c-ae72-c596674491f1) | ![IMG_4197](https://github.com/Yowkees/keyball/assets/50566499/332d062c-01e1-47c3-9b25-d2cd3b078b76) |

大変お手数ですが、ご確認のほどよろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

トラックボール基板はICの方向等合っており半田付けも大丈夫そうですので、ミドル基板から調査したいと思います。
下記添付写真の同色どうしが導通しているかテスターにて確認いただけますでしょうか。
また、ProMicro裏の印をつけた端子が周辺の端子とショートしていないかもご確認ください。

稀にProMicroの初期不良もありますので、左右のProMicroを入れ替えた場合の挙動も確認いただければ幸いです。

以上、お手数ですがご確認よろしくお願いします。

![Keyball39_trackfix](https://github.com/Yowkees/keyball/assets/84209772/a61a8212-38c6-4a11-80b1-9dfd4052d667)

**Comment by torikatsupg** [comment]
コメントありがとうございます。

各色の端子はどれも通電していなさそうでした。

端子周りのショートについて確認し問題ない認識です。
（自信がないためミドル基盤裏の画像を添付いたします。）

![image](https://github.com/Yowkees/keyball/assets/50566499/18362c90-9fd5-4de0-9931-b5e23747bc03)

**Comment by Yowkees** [comment]
同色どうしが導通していないようですとトラックボールは動きませんので、導通するように7Pコンスルーやジャンパー、ProMicroの半田付けを見直してみてください。
それでも導通しない場合には基板の断線の疑いがありますので、同色同士の端子をエナメル線やゴム被覆導線等でショートして修正する必要があります。
お手数ですがご対応お願いいたします。

**Comment by torikatsupg** [comment]
7Pコンスルーの接続不良が原因で、はんだづけをやり直したら無事動きました。
ご教示いただきありがとうございました。
